azurelinuxagent/common/logger.py

```diff
@@ -246,11 +246,17 @@ def _do_rollover(self):
                 if os.path.exists(dst):
                     os.remove(dst)
 
-                # Archive the current log file.
-                with gzip.open(dst, 'wb') as archive:
+                # Archive the current log file. Surrounding by try:except to safegaurd against gzip exceptions, if any.
+                try:
+                    archive = gzip.open(dst, 'wb')
                     with open(self.path, 'rb') as logfile:
                         archive.writelines(logfile)
 
+                    archive.close()
```

Comment:
Safer way of handling opening archive files in Py2.6: https://stackoverflow.com/questions/6086603/statement-with-and-tarfile/6086722#6086722

Reply:
Thanks.